src/v/cluster/scheduling/partition_allocator.cc

```diff
@@ -281,13 +282,14 @@ partition_allocator::allocate(allocation_request request) {
         auto replicas = allocate_partition(
           std::move(p_constraints), request.domain);
         if (!replicas) {
-            return replicas.error();
+            co_return replicas.error();
         }
         assignments.emplace_back(
           _state->next_group_id(), partition_id, std::move(replicas.value()));
+        co_await ss::maybe_yield();
```

Comment:
nit: I wonder what the difference with `ss::coroutine::maybe_yield()` is? Is the coroutine variant more efficient?

Reply:
i will use coroutine one